cmd/kar-controllers/app/discovery/discovery.go

```diff
@@ -0,0 +1,39 @@
+package discovery
+
+import (
+	"k8s.io/klog/v2"
+	"time"
+
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/discovery/cached/memory"
+)
+
+var GlobalCachedDiscoveryClient discovery.CachedDiscoveryInterface
+
+func InitializeGlobalDiscoveryClient(config *rest.Config) error {
+	discoveryClient, err := discovery.NewDiscoveryClientForConfig(config)
+	if err != nil {
+		return err
+	}
+	GlobalCachedDiscoveryClient = memory.NewMemCacheClient(discoveryClient)
+	
+	go startDiscoveryRefreshTicker()
+	
+	return nil
+}
+
+func RefreshDiscoveryCache() {
+	klog.Infof("Invalidating discovery cache")
+	GlobalCachedDiscoveryClient.Invalidate()
```

Comment:
Perhaps we should have some error handling or logging in case any errors arise when attempting to invalidate the discovery cache? WDYT

Reply:
`Invalidate` doesn't return anything so I don't think this is possible

Reply:
Gotcha, you're right! Thanks Kevin.